tests/integration_tests/performance/test_snapshot_ab.py

```diff
@@ -178,32 +178,43 @@ def test_restore_latency(
         }
     )
 
-    ah = test_setup.sample_latency(
-        microvm_factory,
-        guest_kernel_linux_4_14,
-        rootfs,
-        FC_WORKSPACE_DIR / "build" / a_revision,
-    )
-    be = test_setup.sample_latency(
-        microvm_factory,
-        guest_kernel_linux_4_14,
-        rootfs,
-        FC_WORKSPACE_DIR / "build" / b_revision,
-    )
+    if a_revision and b_revision:
+        ah = test_setup.sample_latency(
+            microvm_factory,
+            guest_kernel_linux_4_14,
+            rootfs,
+            FC_WORKSPACE_DIR / "build" / a_revision,
+        )
+        be = test_setup.sample_latency(
+            microvm_factory,
+            guest_kernel_linux_4_14,
+            rootfs,
+            FC_WORKSPACE_DIR / "build" / b_revision,
+        )
 
-    result = check_regression(ah, be, larger_is_better=False)
+        result = check_regression(ah, be, larger_is_better=False)
 
-    metrics.set_property("revision_a", a_revision)
-    metrics.set_property("revision_b", b_revision)
-    metrics.set_property("a_latencies", ah)
-    metrics.set_property("b_latencies", be)
+        metrics.set_property("revision_a", a_revision)
+        metrics.set_property("revision_b", b_revision)
+        metrics.set_property("a_latencies", ah)
+        metrics.set_property("b_latencies", be)
 
-    metrics.put_metric("p_value", result.pvalue, "None")
-    metrics.put_metric("mean_difference", result.statistic, "Milliseconds")
+        metrics.put_metric("p_value", result.pvalue, "None")
+        metrics.put_metric("mean_difference", result.statistic, "Milliseconds")
 
-    metrics.flush()
+        metrics.flush()
 
-    assert (
-        result.pvalue > SIGNIFICANCE_THRESHOLD
-        or result.statistic < sum(ah) / len(ah) * RELATIVE_STRENGTH_THRESHOLD_MS
-    ), f"A/B-testing shows a regression of {result.statistic:.2f}ms with p={result.pvalue}. This means that observing a regression of this magnitude or worse, assuming that performance characteristics did not change across the tested commits, has a probability of {result.pvalue * 100:.2f}%"
+        assert (
+            result.pvalue > SIGNIFICANCE_THRESHOLD
+            or result.statistic < sum(ah) / len(ah) * RELATIVE_STRENGTH_THRESHOLD_MS
+        ), f"A/B-testing shows a regression of {result.statistic:.2f}ms with p={result.pvalue}. This means that observing a regression of this magnitude or worse, assuming that performance characteristics did not change across the tested commits, has a probability of {result.pvalue * 100:.2f}%"
+    else:
+        latencies = test_setup.sample_latency(
+            microvm_factory, guest_kernel_linux_4_14, rootfs
+        )
+
+        metrics.set_property("latency", latencies)
+        metrics.put_metric(
+            "latency_Avg", sum(latencies) / len(latencies), "Milliseconds"
```

Comment:
Could use `statistics.mean(latencies)`. This may be overkill for just a few datapoints, but in general making a numerically robust mean is harder than it seems.

Reply:
Ahhh, neat, I only went looking into the `math` module and got confused when I couldnt find it